src/interfaces/vault.ts

```diff
@@ -680,4 +688,29 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
     };
     return apy;
   }
+
+  /**
+   * Helper function to set the zapper properties on a vault's metadata
+   * @param metadataOverrides the vault metadata overrides
+   * @param vaultTokenMarketData the vault token market data
+   * @returns the updated metadata
+   */
+  mergeZapperProps(
```

Comment:
should this be private function?

Reply:
Maybe. The whole OOP and classes feels a bit too Java-like, in TS/JS I feel we should have a more functional approach. I've left that public so it's easier to test, but in a way that's not specific to vaults, so we could potentially extract this to a helper/util folder to work with zapper? Something like `utils/zapper/mergeZapperProps.ts`? What do you think?

I feel the functions in these classes are doing a bit too much

Reply:
If we end up implementing this change https://github.com/yearn/yearn-sdk/pull/276/files#r844391486 , having this particular function might no longer be needed. I do agree its best to have helper functions when needed, and less tied to its class, unless its very related. Also, those helper functions should aim to be generic if possible.

Reply:
Which change is that? The link too me to a random place

Reply:
Seems link gets outdated when a change is made.